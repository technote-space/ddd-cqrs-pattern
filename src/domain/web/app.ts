import type { AppProps } from 'next/app';

export interface IApp {
  create(): (props: AppProps) => JSX.Element;
}
