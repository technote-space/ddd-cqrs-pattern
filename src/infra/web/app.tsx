import type { IApp } from '$/web/app';
import type { ITheme } from '$/web/theme';
import type { AppProps } from 'next/app';
import { singleton, inject } from 'tsyringe';

@singleton()
export class App implements IApp {
  public constructor(
    @inject('ITheme') private theme: ITheme,
  ) {
  }

  public create(): (props: Pick<AppProps, 'Component' | 'pageProps'>) => JSX.Element {
    // eslint-disable-next-line react/display-name
    return ({ Component, pageProps }: Pick<AppProps, 'Component' | 'pageProps'>) => {
      return this.theme.render({}, <Component {...pageProps} />);
    };
  }
}
