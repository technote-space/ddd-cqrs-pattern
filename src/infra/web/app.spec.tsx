import type { ITheme } from '$/web/theme';
import type { PropsWithChildren } from 'react';
import renderer from 'react-test-renderer';
import { App } from './app';

class TestTheme implements ITheme {
  public render({ children }: PropsWithChildren<any>) {
    return <div className="theme">
      {children}
    </div>;
  }
}

describe('App', () => {
  it('App を生成', () => {
    const mockGetStoreProvider = jest.fn(() => ({ children }: any) => children);
    const app = new App(
      new TestTheme(),
      { getStoreProvider: mockGetStoreProvider } as never,
    );

    const tree = renderer.create(app.create()({
      Component: () => <div>test</div>,
      pageProps: {},
    }));
    expect(tree).toMatchSnapshot();
  });
});
