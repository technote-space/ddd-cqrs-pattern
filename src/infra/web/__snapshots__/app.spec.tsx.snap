// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App を生成 1`] = `
<div
  className="provider2"
>
  <div
    className="provider1"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
